pkg/controller/jobs/pod/pod_webhook.go

```diff
@@ -275,12 +264,16 @@ func (w *PodWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.
 
 	allErrs := jobframework.ValidateJobOnUpdate(oldPod, newPod)
 	allErrs = append(allErrs, validateCommon(newPod)...)
-
-	allErrs = append(allErrs, validation.ValidateImmutableField(podGroupName(newPod.pod), podGroupName(oldPod.pod), groupNameLabelPath)...)
 	allErrs = append(allErrs, validateUpdateForRetriableInGroupAnnotation(oldPod, newPod)...)
 
-	if warn := warningForPodManagedLabel(newPod); warn != "" {
-		warnings = append(warnings, warn)
+	if !newPod.IsSuspended() {
```

Comment:
Ah, I see. I think we can make the condition stronger as suggested above - basically only allow to set it, but not update or remove.

Reply:
Done